modules/openthread/Kconfig.thread

```diff
@@ -61,9 +61,9 @@ config OPENTHREAD_POLL_PERIOD
 
 config OPENTHREAD_MAX_CHILDREN
 	int "The maximum number of children"
-	range 10 511
+	range 1 511
+	default 1 if OPENTHREAD_MTD
```

Comment:
Can you elaborate a bit on why is this needed? Does OT still allocate some children-specific resources even for MTD devices?

Reply:
Yes, for instance the `Metadata` struct of message `Buffer` always contains the `ChildMask mChildMask` which is size dependent on the number of children:

`typedef BitVector<OPENTHREAD_CONFIG_MLE_MAX_CHILDREN> ChildMask;`

There are other cases as well.